### PR TITLE
REF: turn _try_mi into MultiIndex._get_values_for_loc

### PR DESCRIPTION
**Pull request by jbrockmendel**
Discussed in #31640.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
The travis page says its green, not sure why it shows yellow here

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -2326,28 +2339,32 @@ def get_value(self, series, key):
...
+                loc = key
             else:
                 raise
 
+        return self._get_values_for_loc(series, loc, key)
+
+    def _get_values_for_loc(self, series: "Series", loc, key):
+        """
+        Do a positional lookup on the given Series, returning either a scalar
+        or a Series.
```
Comment (jreback) [inline pandas/core/indexes/multi.py]: I would rename to _get_values_for_positional_loc to make this extra clear (I know for indexes we *only* have the concept of positional indexing, but still). follown ok.

also pls type / doc-string as much as possible

**Review by jreback — APPROVED** [review]
a followon request

File: pandas/core/indexes/multi.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/multi.py]: > _get_values_for_positional_loc

good idea, will follow up

**Pull request merged**